### PR TITLE
[New] Add field for Adjusting Position Title in Edit/Create Profile

### LINKED ISSUE #524: Add position title field to Profile
**Is your feature request related to a problem? Please describe.**

Users can manually add their job title to their profile

**Describe the solution you'd like**

- On the Edit profile page in Contact information card, add a field under the first name to manually add their position title.
- In the user profile the title should be displayed just below the user's name

image

image

### PR DESCRIPTION
**Pull request by KateAThornley**
- Added field that autofills with GEDS position title then allows user to change the title in the edit or create primary info forms.
- Allows user to change title to custom title (other than GEDS), by switching language (popup modal upon save to come in next linked PR)
- Added tooltip to header of job title field for explination

### CODE REVIEW

File: services/frontend-v3/src/i18n/en_CA.json [hunk 1]
```diff
@@ -157,6 +157,9 @@
   "profile.branch": "Branch",
   "profile.career.content.name": "Description",
   "profile.career.header.name": "Job title",
+  "profile.career.header.tooltip": "It is recommended to use the information synced from GEDS. However if you edit your job title, in order to display both a customized French and English title, you must change the app language and save each title in the form accordingly.",
```
Comment (BenJeau) [inline services/frontend-v3/src/i18n/en_CA.json]: GEDS has been renamed to GCdirectory
```suggestion
  "profile.career.header.tooltip": "It is recommended to use the information synced from GCdirectory. However if you edit your job title, in order to display both a customized French and English title, you must change the app language and save each title in the form accordingly.",
```

File: services/frontend-v3/src/i18n/en_CA.json [hunk 2]
```diff
@@ -157,6 +157,9 @@
...
+  "profile.change.career.title.modal": "French job title",
+  "profile.change.career.title.desc.modal": "Enter a French translation of your job title to be displayed on your French profile, or if you've synced with GEDS and would like to keep that title, leave this field blank. If you leave this field blank and have not synced your job title from GEDS then your job title on your French profile will be empty.",
```
Comment (BenJeau) [inline services/frontend-v3/src/i18n/en_CA.json]:
```suggestion
  "profile.change.career.title.desc.modal": "Enter a French translation of your job title to be displayed on your French profile, or if you've synced with GCdirectory and would like to keep that title, leave this field blank. If you leave this field blank and have not synced your job title from GCdirectory then your job title on your French profile will be empty.",
```

File: services/frontend-v3/src/i18n/fr_CA.json [hunk 3]
```diff
@@ -157,6 +157,9 @@
   "profile.branch": "Branche",
   "profile.career.content.name": "Description",
   "profile.career.header.name": "Poste de travail",
+  "profile.career.header.tooltip": "Il est recommandé d'utiliser les informations synchronisées depuis SAGE. Cependant, si vous modifiez l'intitulé de votre poste, afin d'afficher un titre personnalisé en français et en anglais, vous devez changer la langue de l'application et enregistrer chaque titre dans le formulaire en conséquence.",
```
Comment (BenJeau) [inline services/frontend-v3/src/i18n/fr_CA.json]:
```suggestion
  "profile.career.header.tooltip": "Il est recommandé d'utiliser les informations synchronisées depuis GCannuaire. Cependant, si vous modifiez le titre de votre poste, afin d'afficher un titre personnalisé en français et en anglais, vous devez changer la langue de l'application et enregistrer chaque titre dans le formulaire en conséquence.",
```

File: services/frontend-v3/src/i18n/fr_CA.json [hunk 4]
```diff
@@ -157,6 +157,9 @@
...
+  "profile.change.career.title.modal": "Poste de travail en anglais",
```
Comment (BenJeau) [inline services/frontend-v3/src/i18n/fr_CA.json]:
```suggestion
  "profile.change.career.title.modal": "Titre de poste de travail en anglais",
```

File: services/frontend-v3/src/i18n/fr_CA.json [hunk 5]
```diff
@@ -157,6 +157,9 @@
...
+  "profile.change.career.title.desc.modal": "Saisissez une traduction anglaise de l'intitulé de votre poste à afficher sur votre profil français, ou si vous avez synchronisé avec SAGE et que vous souhaitez conserver ce titre, laissez ce champ vide. Si vous laissez ce champ vide et n'avez pas synchronisé le titre de votre poste depuis SAGE, le titre de votre poste sur votre profil anglais sera vide.",
```
Comment (BenJeau) [inline services/frontend-v3/src/i18n/fr_CA.json]:
```suggestion
  "profile.change.career.title.desc.modal": "Saisissez une traduction anglaise du titre de votre poste à afficher sur votre profil français, ou si vous avez synchronisé avec GCannuaire et que vous souhaitez conserver ce titre, laissez ce champ vide. Si vous laissez ce champ vide et n'avez pas synchronisé le titre de votre poste depuis GCannuaire, le titre de votre poste sur votre profil anglais sera vide.",
```

File: services/frontend-v3/src/components/sideNav/logo_v2 2.svg [hunk 6]
```diff
@@ -0,0 +1 @@
...rm="matrix(1.00137 0 -0 1 314.001 316.775)"><use width="100%" height="100%" xlink:href="#img7" transform="scale(0.997444 0.997444)"></use></g></g></g><path d="M308.21 375.538C308.21 346.617 331.655 323.172 360.576 323.172 389.497 323.172 412.942 346.617 412.942 375.538 412.942 404.458 389.497 427.903 360.576 427.903 331.655 427.903 308.21 404.458 308.21 375.538Z" fill="#08BAB3" fill-rule="evenodd" transform="matrix(1.00137 0 -0 1 10.3636 -6.39652)"/><g clip-path="url(#clip12)"><g clip-path="url(#clip13)"><g clip-path="url(#clip14)" transform="matrix(1.00137 0 -0 1 386.914 316.775)"><use width="100%" height="100%" xlink:href="#img7" transform="scale(0.997444 0.997444)"></use></g></g></g><path d="M381.023 375.538C381.023 346.617 404.468 323.172 433.389 323.172 462.31 323.172 485.755 346.617 485.755 375.538 485.755 404.458 462.31 427.903 433.389 427.903 404.468 427.903 381.023 404.458 381.023 375.538Z" fill="#007572" fill-rule="evenodd" transform="matrix(1.00137 0 -0 1 10.3636 -6.39652)"/><text fill="#FFFFFF" font-family="Arial,Arial_MSFontService,sans-serif" font-weight="400" font-size="160" transform="matrix(0.998808 0 0 0.997444 511.98 398)">UpSkill</text></g></g></g></g></g></svg>
```
Comment (BenJeau) [inline services/frontend-v3/src/components/sideNav/logo_v2 2.svg]: What is this file?

**Review by BenJeau — COMMENTED** [review]
Looks good, just tested it locally and it seems like it doesn't save in french? I save my jobtitle in english, but didn't seem to save in french. not sure if its because of the API or the frontend, if you don't mind, right now the backend only returns the jobtitle in the specified language, but I can push a small fix so it returns the english one if the french is not specified and vice-versa

**Comment by KateAThornley** [comment]
> Looks good, just tested it locally and it seems like it doesn't save in french? I save my jobtitle in english, but didn't seem to save in french. not sure if its because of the API or the frontend, if you don't mind, right now the backend only returns the jobtitle in the specified language, but I can push a small fix so it returns the english one if the french is not specified and vice-versa

Strange that it's not saving in French for you. Right now if I change the language to French it brings up and saves the French job title, but I just relaized it doesnt automatically fill (you need to reload if youre changing it on the same attempt that you change the other language's title). Feel free to commit the fix- it is greatly appreciated!

**Comment by dynamic11** [comment]
is this PR ready to merge? @KateAThornley

**Comment by KateAThornley** [comment]
> is this PR ready to merge? @KateAThornley

Good morning! Going to resolve conflicts then yes.

File: services/backend/package.json [hunk 7]
```diff
@@ -13,9 +13,7 @@
     "lint": "./node_modules/.bin/eslint src",
     "seed": "node src/database/seeds",
     "studio": "cd src/database && prisma studio --experimental",
-    "test": "node ./tests/setupDatabase && cross-env NODE_ENV=test DATABASE_URL=$TEST_DATABASE_URL jest",
-    "snyk-protect": "snyk protect",
-    "prepare": "yarn run snyk-protect"
```
Comment (dynamic11) [inline services/backend/package.json:18]: you shouldn't delete these

File: services/backend/package.json [hunk 8]
```diff
@@ -33,13 +31,12 @@
...
     "fuse.js": "^6.4.1",
     "keycloak-connect": "^11.0.0",
-    "lodash": "^4.17.20",
+    "lodash": "^4.17.19",
     "moment": "^2.27.0",
     "redis": "^3.0.2",
     "swagger-jsdoc": "^4.0.0",
     "swagger-ui-express": "^4.1.4",
-    "validator": "^13.1.1",
-    "snyk": "^1.374.0"
```
Comment (dynamic11) [inline services/backend/package.json:42]: you shouldn't delete these

File: services/backend/package.json [hunk 9]
```diff
@@ -33,13 +31,12 @@
     "express-validator": "^6.6.1",
     "fuse.js": "^6.4.1",
     "keycloak-connect": "^11.0.0",
-    "lodash": "^4.17.20",
+    "lodash": "^4.17.19",
```
Comment (dynamic11) [inline services/backend/package.json:34]: you shouldn't change this

File: services/backend/src/config.js [hunk 10]
```diff
@@ -69,7 +69,7 @@ const development = {
     "/auth/realms/individual/protocol/openid-connect/token",
   GEDSAPIURL:
     GEDSAPIURL ||
-    "https://geds-sage-ssc-spc-apicast-production.api.canada.ca/gapi/v2/",
+    "https://geds-ssc-spc-apicast-production.api.canada.ca/gapi/v2/",
```
Comment (dynamic11) [inline services/backend/src/config.js:72]: why has this changed? Im just curious

File: services/backend/src/core/connections/connections.js [hunk 11]
```diff
@@ -81,11 +74,9 @@ async function removeConnection(request, response) {
 async function getConnectionById(request, response) {
   try {
     validationResult(request).throw();
-
-    const { id } = request.params;
-    const userId = getKeycloakUserId(request);
-
-    if (userId && id) {
```
Comment (dynamic11) [inline services/backend/src/core/connections/connections.js:88]: It think this was removed due to merge conflict. this was changed 7 days ago by ben so i dont think it shoul dbe changed back

**Review by dynamic11 — CHANGES_REQUESTED** [review]
I think something went wrong when you updated your pranch from development

File: services/backend/src/config.js (hunk 10 above)
Comment (BenJeau) [reply services/backend/src/config.js:72]: this should be changed back, the link with `sage` is the correct one

**Pull request closed without merging**